Add libwebrtc specific WebCore RTCStat object constructors

youennf opened the pull request:
#### d7759dc9cbb90b3fda92a54521c132c18cc5a0df
<pre>
Add libwebrtc specific WebCore RTCStat object constructors
<a href="https://rdar.apple.com/129205417">rdar://129205417</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275096">https://bugs.webkit.org/show_bug.cgi?id=275096</a>

Reviewed by Jean-Yves Avenard.

Refactoring to do more initialization in constructors.
This makes it easier to initialize all values and will help code analysis.

* Source/WebCore/Modules/mediastream/RTCStatsReport.h:
(WebCore::RTCStatsReport::InboundRtpStreamStats::InboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::DataChannelStats::DataChannelStats): Deleted.
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats): Deleted.
(WebCore::RTCStatsReport::CertificateStats::CertificateStats): Deleted.
(WebCore::RTCStatsReport::CodecStats::CodecStats): Deleted.
(WebCore::RTCStatsReport::TransportStats::TransportStats): Deleted.
(WebCore::RTCStatsReport::AudioPlayoutStats::AudioPlayoutStats): Deleted.
(WebCore::RTCStatsReport::PeerConnectionStats::PeerConnectionStats): Deleted.
(WebCore::RTCStatsReport::AudioSourceStats::AudioSourceStats): Deleted.
(WebCore::RTCStatsReport::VideoSourceStats::VideoSourceStats): Deleted.
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
(WebCore::RTCStatsReport::Stats::Stats):
(WebCore::RTCStatsReport::RtpStreamStats::RtpStreamStats):
(WebCore::RTCStatsReport::ReceivedRtpStreamStats::ReceivedRtpStreamStats):
(WebCore::RTCStatsReport::InboundRtpStreamStats::InboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats):
(WebCore::RTCStatsReport::SentRtpStreamStats::SentRtpStreamStats):
(WebCore::convertQualityLimitationReason):
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats):
(WebCore::RTCStatsReport::DataChannelStats::DataChannelStats):
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats):
(WebCore::RTCStatsReport::IceCandidateStats::IceCandidateStats):
(WebCore::RTCStatsReport::CertificateStats::CertificateStats):
(WebCore::RTCStatsReport::CodecStats::CodecStats):
(WebCore::convertIceRole):
(WebCore::convertDtlsRole):
(WebCore::RTCStatsReport::TransportStats::TransportStats):
(WebCore::RTCStatsReport::PeerConnectionStats::PeerConnectionStats):
(WebCore::RTCStatsReport::MediaSourceStats::MediaSourceStats):
(WebCore::RTCStatsReport::AudioSourceStats::AudioSourceStats):
(WebCore::RTCStatsReport::AudioPlayoutStats::AudioPlayoutStats):
(WebCore::RTCStatsReport::VideoSourceStats::VideoSourceStats):
(WebCore::addToStatsMap):
(WebCore::initializeRTCStatsReportBackingMap):
(WebCore::fillRTCStats): Deleted.
(WebCore::fillRtpStreamStats): Deleted.
(WebCore::fillReceivedRtpStreamStats): Deleted.
(WebCore::fillInboundRtpStreamStats): Deleted.
(WebCore::fillRemoteInboundRtpStreamStats): Deleted.
(WebCore::fillSentRtpStreamStats): Deleted.
(WebCore::qualityLimitationReason): Deleted.
(WebCore::fillOutboundRtpStreamStats): Deleted.
(WebCore::fillRemoteOutboundRtpStreamStats): Deleted.
(WebCore::fillRTCDataChannelStats): Deleted.
(WebCore::fillRTCIceCandidatePairStats): Deleted.
(WebCore::fillRTCIceCandidateStats): Deleted.
(WebCore::fillRTCCertificateStats): Deleted.
(WebCore::fillRTCCodecStats): Deleted.
(WebCore::iceRole): Deleted.
(WebCore::dtlsRole): Deleted.
(WebCore::fillRTCTransportStats): Deleted.
(WebCore::fillRTCPeerConnectionStats): Deleted.
(WebCore::fillRTCMediaSourceStats): Deleted.
(WebCore::fillRTCAudioSourceStats): Deleted.
(WebCore::fillRTCAudioPlayoutStats): Deleted.
(WebCore::fillRTCVideoSourceStats): Deleted.

Canonical link: <a href="https://commits.webkit.org/279731@main">https://commits.webkit.org/279731@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03250f464d8c3dbba4c29f77f290d082f32c8101

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54349 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6905 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57629 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5081 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56652 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41279 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/5029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44010 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3391 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56443 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47064 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25146 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/28735 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4386 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3224 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4596 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59219 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/29568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/5029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/30728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47157 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50798 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11836 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/31702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/30514 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 55;
```
Shouldn't be in there

youennf replied:
Will remove but it will probably keep coming...

jyavenard commented on Source/WebCore/Modules/mediastream/RTCStatsReport.h:
```diff
@@ -61,26 +87,44 @@ class RTCStatsReport : public RefCounted<RTCStatsReport> {
+#endif
+
         double timestamp { 0 };
         Type type;
```
Shouldn't this be initiises too?

youennf replied:
The constructor should do it

youennf replied:
Plan is to remove the default constructor so that Stats(Type,...) is responsible for it.

jyavenard commented on Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
```diff
@@ -648,98 +632,75 @@ static inline void fillRTCAudioSourceStats(RTCStatsReport::AudioSourceStats& sta
-            fillRTCAudioPlayoutStats(stats, static_cast<const webrtc::RTCAudioPlayoutStats&>(rtcStats));
-            report.set<IDLDOMString, IDLDictionary<RTCStatsReport::AudioPlayoutStats>>(stats.id, WTFMove(stats));
-        }
+        if (rtcStats.type() == webrtc::RTCInboundRtpStreamStats::kType)
```
This seems less than ideal

jyavenard replied:
Switch maybe?

youennf replied:
type() is a const char*, it should indeed be migrated to an enum in libwebrtc, in the meantime, we do not have the choice really.

youennf committed:
Add libwebrtc specific WebCore RTCStat object constructors
rdar://129205417
https://bugs.webkit.org/show_bug.cgi?id=275096

Reviewed by Jean-Yves Avenard.

Refactoring to do more initialization in constructors.
This makes it easier to initialize all values and will help code analysis.

* Source/WebCore/Modules/mediastream/RTCStatsReport.h:
(WebCore::RTCStatsReport::InboundRtpStreamStats::InboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats): Deleted.
(WebCore::RTCStatsReport::DataChannelStats::DataChannelStats): Deleted.
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats): Deleted.
(WebCore::RTCStatsReport::CertificateStats::CertificateStats): Deleted.
(WebCore::RTCStatsReport::CodecStats::CodecStats): Deleted.
(WebCore::RTCStatsReport::TransportStats::TransportStats): Deleted.
(WebCore::RTCStatsReport::AudioPlayoutStats::AudioPlayoutStats): Deleted.
(WebCore::RTCStatsReport::PeerConnectionStats::PeerConnectionStats): Deleted.
(WebCore::RTCStatsReport::AudioSourceStats::AudioSourceStats): Deleted.
(WebCore::RTCStatsReport::VideoSourceStats::VideoSourceStats): Deleted.
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
(WebCore::RTCStatsReport::Stats::Stats):
(WebCore::RTCStatsReport::RtpStreamStats::RtpStreamStats):
(WebCore::RTCStatsReport::ReceivedRtpStreamStats::ReceivedRtpStreamStats):
(WebCore::RTCStatsReport::InboundRtpStreamStats::InboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats):
(WebCore::RTCStatsReport::SentRtpStreamStats::SentRtpStreamStats):
(WebCore::convertQualityLimitationReason):
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats):
(WebCore::RTCStatsReport::DataChannelStats::DataChannelStats):
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats):
(WebCore::RTCStatsReport::IceCandidateStats::IceCandidateStats):
(WebCore::RTCStatsReport::CertificateStats::CertificateStats):
(WebCore::RTCStatsReport::CodecStats::CodecStats):
(WebCore::convertIceRole):
(WebCore::convertDtlsRole):
(WebCore::RTCStatsReport::TransportStats::TransportStats):
(WebCore::RTCStatsReport::PeerConnectionStats::PeerConnectionStats):
(WebCore::RTCStatsReport::MediaSourceStats::MediaSourceStats):
(WebCore::RTCStatsReport::AudioSourceStats::AudioSourceStats):
(WebCore::RTCStatsReport::AudioPlayoutStats::AudioPlayoutStats):
(WebCore::RTCStatsReport::VideoSourceStats::VideoSourceStats):
(WebCore::addToStatsMap):
(WebCore::initializeRTCStatsReportBackingMap):
(WebCore::fillRTCStats): Deleted.
(WebCore::fillRtpStreamStats): Deleted.
(WebCore::fillReceivedRtpStreamStats): Deleted.
(WebCore::fillInboundRtpStreamStats): Deleted.
(WebCore::fillRemoteInboundRtpStreamStats): Deleted.
(WebCore::fillSentRtpStreamStats): Deleted.
(WebCore::qualityLimitationReason): Deleted.
(WebCore::fillOutboundRtpStreamStats): Deleted.
(WebCore::fillRemoteOutboundRtpStreamStats): Deleted.
(WebCore::fillRTCDataChannelStats): Deleted.
(WebCore::fillRTCIceCandidatePairStats): Deleted.
(WebCore::fillRTCIceCandidateStats): Deleted.
(WebCore::fillRTCCertificateStats): Deleted.
(WebCore::fillRTCCodecStats): Deleted.
(WebCore::iceRole): Deleted.
(WebCore::dtlsRole): Deleted.
(WebCore::fillRTCTransportStats): Deleted.
(WebCore::fillRTCPeerConnectionStats): Deleted.
(WebCore::fillRTCMediaSourceStats): Deleted.
(WebCore::fillRTCAudioSourceStats): Deleted.
(WebCore::fillRTCAudioPlayoutStats): Deleted.
(WebCore::fillRTCVideoSourceStats): Deleted.

Canonical link: https://commits.webkit.org/279731@main
